.github/sync_labels.py

```diff
@@ -408,7 +408,17 @@ def get_commits(self):
             return self._commits
 
         self._commits = self.view('commits')
-        self._commit_date = max( com['committedDate'] for com in self._commits )
+
+        # ignore merge commits with the develop branch for commit_date if needs_work is set
+        date_commits = list(self._commits)
+        if Status.needs_work.value in self.get_labels():
+            for com in date_commits:
+                message =  com['messageHeadline']
+                if message.startswith('Merge') and message.find('develop') > 0:
```

Comment:
```suggestion
                if message.startswith("Merge branch 'develop' into "):
```
Would this work? Since it's more specific it's less likely to have a false positive.

Reply:
Never mind, I see you already mentioned this. You can ignore this suggestion if you think people changing the merge commit message could be an issue.

Reply:
There are also default messages that wouldn't match, for example in #40448: `Merge remote-tracking branch 'upstream/develop' into remove-incremental`